discovery-provider/src/tasks/index.py

```diff
@@ -200,8 +221,12 @@ def fetch_tx_receipts(self, block_transactions):
     num_processed_txs = len(block_tx_with_receipts.keys())
     num_submitted_txs = len(block_transactions)
     if num_processed_txs != num_submitted_txs:
-        raise Exception(
-            f"index.py | fetch_tx_receipts Expected ${num_submitted_txs} received {num_processed_txs}"
+        raise IndexingError(
```

Comment:
Not sure if this should raise an IndexingError because a failure here should always be retried, curious for your thinking there. I don't think the IndexingError will cause harm though. Is it worth using this helper? `create_and_raise_indexing_error`

Reply:
I factored this code out into a separate fn and wasn't aware of the original intent behind it. Throwing an IndexingError instead of an Exception here was because the test case I wrote expected an IndexingError. I think either way this exception will percolate up - the distinction is partly semantic and partly in whether the error gets cached in redis.